### DISCUSSION TITLE
Drawer tooltip remains when drawer collapses

### DISCUSSION THREAD

**Discussion by CrawfordW** [quasarframework/quasar#9521] [General - Components / Directives / etc]
I have a left drawer with buttons that trigger an activity in the main page and, if the drawer is in overlay mode, cause the drawer to collapse so the main page can be active. 
Each button has a tooltip which is generally visible when the button is clicked. Sometimes, the tooltip remains on the screen after the drawer has collapsed, and there's no way to make it disappear except to open the drawer again and hover over the same button.
Is there a way to prevent this behaviour? I'd like to ensure any tooltips from the drawer disappear when the drawer collapses.
Maybe this is a quasar bug?
The problem only happens sometimes - I can't reliably repeat it.

**Comment by ontwikkelfabriek** [comment]
@CrawfordW 

Without a codepen that reproduces your problem it's nearly impossible to tell what is going on. 

Btw are you using qtooltip?

**Reply by CrawfordW** [reply]
Yes it's a custom widget that includes a q-btn and q-tooltip.
I think the whole thing is too complex to try and make a codepen - I was hoping maybe there's some trick to force a tooltip to go away when the thing that launched it is no longer being rendered - it's on a keep-alive tab in an overlay drawer that get's collapsed when the button @click method runs. So the tab with the panel is still alive but not rendered - tooltip should also go away.
It usually works fine, but occasionally the tooltip stays when the drawer collapses and then there's no way to get rid of it. I think this may be a quasar bug unless there's something I'm supposed to do with the tooltip that I'm not aware of.

**Reply by ontwikkelfabriek** [reply]
@CrawfordW 

You could try to put a `ref` on the Qtooltip\Button\Widget and do a `v-if/v-show` binding on the state of the drawer or the "click" of the button to make it disappear.

**Reply by CrawfordW** [reply]
Sounds interesting! I'll give that a try. Thanks!

**Reply by CrawfordW** [reply]
Seems to work! Good suggestion!
Here's what I did in my button widget...

<q-tooltip v-if="showTip"

mounted () {
    this.showTip = true
}

onClick () {
      this.showTip = false
      this.$emit('click', this.name)
      setTimeout(() => {
        this.showTip = true
      }, 1000)
}
